server/index.js

```diff
@@ -2,15 +2,38 @@ require('dotenv/config');
 const express = require('express');
 const staticMiddleware = require('./static-middleware');
 const errorMiddleware = require('./error-middleware');
+const axios = require('axios');
+const crypto = require('crypto');
 
 const app = express();
 
-app.use(staticMiddleware);
+const API_PUBLIC_KEY = 'b8483fd7fba99cd20a9fefc4e5106f88';
+const API_PRIVATE_KEY = 'c2746ff73c66112e104538fe16622cbb21205d8f';
+
+app.get('/marvel/:characterName', (req, res) => {
+
+  const timestamp = Date.now().toString();
+  const hash = crypto.createHash('md5').update(timestamp + API_PRIVATE_KEY + API_PUBLIC_KEY).digest('hex');
+  const characterName = req.params.characterName;
+  const url = `https://gateway.marvel.com/v1/public/characters?apikey=${API_PUBLIC_KEY}&ts=${timestamp}&hash=${hash}&name=${encodeURIComponent(characterName)}`;
+
+  axios.get(url)
+    .then(response => {
+      const data = response.data.data;
+      const character = data.results[0]; // Assuming the search only returned one result
+      const characterName = character.name;
+      const characterDescription = character.description;
+      const characterThumbnailUrl = `${character.thumbnail.path}.${character.thumbnail.extension}`;
+      res.status(200).send(`Name: ${characterName}, Description: ${characterDescription}, Thumbnail URL: ${characterThumbnailUrl}`);
+    })
+    .catch(error => {
+      console.error(error);
+    });
 
-app.get('/api/hello', (req, res) => {
-  res.json({ hello: 'world' });
 });
 
+app.use(staticMiddleware);
```

Comment:
This one usually goes at the top of the middleware, though I'm not sure it matters.